components/zcash_protocol/src/local_consensus.rs

```diff
@@ -43,6 +43,54 @@ pub struct LocalNetwork {
     pub z_future: Option<BlockHeight>,
 }
 
+impl LocalNetwork {
+    /// Construct a new `LocalNetwork` where all network upgrade activation heights are specified.
+    ///
+    /// Sequential network upgrades may share the same activation height but this constructor will
+    /// panic if the order in which network upgrades activate is changed.
+    pub fn new(
+        overwinter: u64,
+        sapling: u64,
+        blossom: u64,
+        heartwood: u64,
+        canopy: u64,
+        nu5: u64,
```

Comment:
I just realised that all of these arguments are `u64`. Two questions:
- Why not `BlockHeight`?
- If not `BlockHeight`, why `u64`? The internal casts to `u32` will silently truncate the heights, which is undocumented and confusing. Additionally, these truncations occur after the assertion, which means that a caller can bypass the ordering requirement by setting the high bits of the heights.

The latter question is blocking; these should be `u32` at a minimum, and I don't see a good reason for these to not be `BlockHeight`.